tests/testthat/test-unnecessary_nested_if_linter.R

```diff
@@ -174,7 +183,9 @@ test_that("unnecessary_nested_if_linter skips allowed usages", {
 
 test_that("unnecessary_nested_if_linter blocks disallowed usages", {
   lint_message <- rex::rex("Don't use nested `if` statements")
-  linter <- unnecessary_nested_if_linter()
+  expect_warning({
```

Comment:
Which is better, to test for the warning multiple times, or just once and then use `suppressWarnings()`?

Reply:
I followed the approach taken for extraction_operator_linter(). Other than that I'm indifferent

Reply:
Consistency should be the way to go here.
I don't have a preference either.